common-libs/common/src/shared/DidResolver.ts

```diff
@@ -17,42 +24,70 @@ const getService = (options: ConstructorOptions) => {
   if (!services.has(cacheKey)) {
     services.set(cacheKey, {
       service: new RegistryApiService(options),
-      cache: new Map(),
+      cache: options.useCache
+        ? new LRUCache({
+            max: options.cacheMaxSize ?? DEFAULT_CACHE_MAX_SIZE,
+            ttl: (options.cacheTtlInMin ?? DEFAULT_CACHE_TTL_IN_MIN) * 60 * 1000,
+            allowStale: true,
+          })
+        : undefined,
```

Comment:
I see the error in case of `options.useCache` is false.
cache is undefined and on line 41, the error will happen. Am I right? If so, please handle the situation when cache is undefined, and test it